### PR TITLE
Syntax factoring

### PR DESCRIPTION
**Pull request by felixleo22**

### CODE REVIEW

File: bin/update.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+const verbose = async () => {
+  let index = 0;
+  let state;
+  state = await getState('', true);
+  let { steps } = state;
+  steps.forEach(async (step) => {
+    if (step?.percent === 100) {
+      createCliProgress(100, step?.task, step?.file);
+      console.log('\r');
```
Comment (nojhamster) [inline bin/update.js]:
```suggestion
      console.log();
```

`console.log()` already prints to `stdout` with a `new line`. Additionally, the proper character for line breaks is `\n` (`\r` is a `carriage return`).

File: bin/update.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        }
+        steps = state?.steps;
+        if (Array.isArray(steps)) {
+          steps = steps.filter((x) => x.task !== 'getChangefiles');
+          latestStep = steps[index];
+          bar.update(Number(latestStep.percent));
+        }
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+      console.log('\r');
```
Comment (nojhamster) [inline bin/update.js]:
```suggestion
      console.log();
```

`console.log()` already prints to `stdout` with a `new line`. Additionally, the proper character for line breaks is `\n` (`\r` is a `carriage return`).

File: bin/update.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    logger.error(err);
+    process.exit(1);
+  }
+  return res?.data?.state || res?.data;
+};
+
+const verbose = async () => {
+  let index = 0;
+  let state;
+  state = await getState('', true);
```
Comment (nojhamster) [inline bin/update.js]:
```suggestion
  let state = await getState('', true);
```

File: cmds/update.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+ * @param {Boolean} option.list -L --list | list of snapshots installed on ezunpaywall
+ * @param {Boolean} option.index -I --index | name of the index to which the data is inserted
+ */
+const updateJobFile = async (option) => {
+  const ezunpaywall = await connection();
+  const config = await getConfig();
+
+  if (option.file) {
+    const pattern = /^[a-zA-Z0-9_.-]+(.gz)$/;
+    if (!pattern.test(option.file)) {
```
Comment (nojhamster) [inline cmds/update.js]: You can make it simpler. Also don't forget that `.` is a special character in regular expressions.

```suggestion
    if (!/\.gz$/i.test(option.file)) {
```

File: cmds/update.js [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+      logger.error('Only ".gz" files are accepted');
+      process.exit(1);
+    }
+  }
+
+  if (option.limit && option.offset) {
+    if (Number(option.limit) <= Number(option.offset)) {
+      logger.error('Limit cannot be low than offset or 0');
```
Comment (nojhamster) [inline cmds/update.js]:
```suggestion
      logger.error('Limit cannot be lower than offset or 0');
```

File: cmds/update.js [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+  if (option.interval) {
+    const intervals = ['week', 'day'];
+    if (!intervals.includes(option.interval)) {
+      logger.error(`${option.interval} is not accepted, only 'week' and 'day' are accepted`);
+      process.exit(1);
+    }
+  }
+
+  if (new Date(option.startDate).getTime() > Date.now()) {
+    logger.error('startDate cannot be in the futur');
```
Comment (nojhamster) [inline cmds/update.js]:
```suggestion
    logger.error('startDate cannot be in the future');
```

File: cmds/update.js [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+    if (new Date(option.endDate).getTime() < new Date(option.startDate).getTime()) {
+      logger.error('endDate cannot be lower than startDate');
+      process.exit(1);
+    }
+  }
+
+  const pattern = /^\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])$/;
+
+  if (option.startDate && !pattern.test(option.startDate)) {
+    logger.error('startDate are in wrong format, required YYYY-mm-dd');
```
Comment (nojhamster) [inline cmds/update.js]:
```suggestion
    logger.error('startDate is in wrong format, required YYYY-mm-dd');
```

File: cmds/update.js [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+    process.exit(1);
+  }
+
+  if (option.endDate && !pattern.test(option.endDate)) {
+    logger.error('endDate are in wrong format, required YYYY-mm-dd');
```
Comment (nojhamster) [inline cmds/update.js]:
```suggestion
    logger.error('endDate is in wrong format, required YYYY-mm-dd');
```

File: bin/update.js [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+    logger.warn('No snapshots are installed in ezunpaywall, it is recommended to install the large snapshot available every 6 months before launching the updates');
+    logger.warn('you can force the update with --force');
+    process.exit(0);
+  }
+
+  const report = await getReport('', { latest: true });
+
+  if (report) {
+    const tasks = report.steps.filter((x) => x.task === 'insert');
+    const [task] = tasks.reverse();
```
Comment (nojhamster) [inline bin/update.js]:
```suggestion
    const task = report.steps.filter((x) => x.task === 'insert').pop();
```

File: cmds/apikey.js [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+  }
+
+  let data;
+
+  try {
+    data = await fs.readFile(filepath);
+    data = JSON.parse(data);
+  } catch (err) {
+    logger.error(err);
+  }
```
Comment (nojhamster) [inline cmds/apikey.js]:
```suggestion
  let data;

  try {
    const content = await fs.readFile(filepath);
    data = JSON.parse(content);
  } catch (err) {
    if (err.code === 'ENOENT') {
      logger.error(`[${filepath}] not found`);
    } else {
      logger.error(err);
    }
    process.exit(1);
  }
```

File: cmds/update.js [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+      choices: reports,
+      message: 'files',
+      default: reports.slice(),
+      source: (answersSoFar, input) => new Promise((resolve) => {
+        const result = reports.files
+          .filter((file) => file.toLowerCase().includes(input.toLowerCase()));
+        resolve(result);
+      }),
+    }]);
+    filename = filename.files;
```
Comment (nojhamster) [inline cmds/update.js]:
```suggestion
    const { files: filename } = await inquirer.prompt([{
      type: 'list',
      pageSize: 5,
      name: 'files',
      choices: reports,
      message: 'files',
      default: reports.slice(),
      source: (answersSoFar, input) => new Promise((resolve) => {
        const result = reports.files
          .filter((file) => file.toLowerCase().includes(input.toLowerCase()));
        resolve(result);
      }),
    }]);
```

File: cmds/update.js [hunk 12]
```diff
@@ -0,0 +1,321 @@
...
+    logger.info('An update is being done');
+    if (option.verbose) {
+      await verbose();
+    } else {
+      const state = await getState('', true);
+      console.log(JSON.stringify(state, null, 2));
+      process.exit(0);
+    }
+  }
+  process.exit(0);
```
Comment (nojhamster) [inline cmds/update.js]: Unnecessary `else` block

```suggestion
  }
  
  logger.info('An update is in progress');
  
  if (option.verbose) {
    await verbose();
  } else {
    const state = await getState('', true);
    console.log(JSON.stringify(state, null, 2));
  }
  process.exit(0);
```

File: bin/update.js (hunk 9 above)
Comment (felixleo22) [reply bin/update.js]: Maybe we should discuss about that

**Pull request merged**